modules/web/src/de/diedavids/cuba/dataimport/web/importwizard/import-wizard.xml

```diff
@@ -74,11 +74,17 @@
                  expand="mapAttributesTable"
                  margin="true"
                  spacing="true">
-                <lookupField id="entityLookup"
-                             caption="Entity"
-                             datasource="importConfigurationDs"
-                             property="entityClass"
-                             width="100%"/>
+                <vbox>
+                    <lookupField id="entityLookup"
+                                 caption="Entity"
+                                 datasource="importConfigurationDs"
+                                 property="entityClass"
+                                 width="100%"/>
+                    <lookupField id="configLookup"
```

Comment:
I don't fully understand your idea here behind the import configuration. Here's the thing: Either the user decides to use an existing import configuration. Then all of the steps up until the preview are not relevant anymore. They need to be skipped. Because the user does not know what happens if in the entity attribute table will be changed although a import configuration was selected.

So for me it is an either / or approach: Either select a import configuration or configure the import configuration (and potentially store it for later use).

Reply:
I see your point of view.

The solution I proposed is based on the freedom of selecting a configuration or not. First you select the entity to upload, then if there is any configuration in the DB related to that entity (main rule), then this lookup list will show and User will be able to select, otherwise can continue modifying.
But at the same time, the user can select the config and still overwrite it, but the config will not be modified in the DB, which is cool.

To simplify, you can import based on existing config or use the default one which was auto generated.
I think this solution gives more possibilities to the user, adding some message information will be very helpful to the user.

Reply:
I just tried what you described. You are right. It is possible to select a import config, then this will be loaded. It is adjustable then and with that can be used as a template.

Unfortunately this is pretty unintuitive to understand right now, so I'll not merge it as is.

I'm not sure where this feature of reusing existing import configs is going. The main reason is that the main user of the import wizard. My assumption is that the people that actually want to import the data are normally not the people that are capable of writing groovy scripts.
So most likely the configs will be pre-defined by administrators / developers.
Afterwards they are just "used" by the users. This means that then the easiest path to get from "I want to import" to "I imported successfully" should be preferred. So all the config would need to go out of the wizard.

In this respect your features is just useful for the devs / admins but not for the users, which should be the main user of that screen.

I would suggest that you pull out this part of the PR so that it can get merged more quickly.

We can then in a extra issue & PR talk about that topic in more detail.

Reply:
To be honest the Import Wizard is for devs/admins/master users. It's the same for all tools (MSSQL, SQL dev... ) and the wizard is in the Admin section naturally
Typical user will be asked to use the Import button.. basicly the #32
For admin/dev/master-user : Wizard

Anyway if you think we should avoid config part for now then I created a branch https://github.com/genthalili/cuba-component-data-import/tree/0.4.0-B

You can create remove the code from `ImportWizard` and contribute directly into my repo.

Reply:
I don't think so. The reason is that I see the current ImportWizard not really as the end of the game. As you might have seen in #76 this is actually my main goal for the users. Not configure anything but still have a wizard with preview, summary etc.

Eventually I think the current Import Wizard can go away, since the normal create operation for an import config is actually quite similar to the current wizard.

I started out with the wizard as a hybrid solution, but I think this is not optimal...